### PR TITLE
fluent: docs: style updates for what is page

### PR DESCRIPTION
**Pull request by esmerel**
This PR does some basic style updates. 

There are still some vale and markdownlint suggestions, but a couple of them are spurious and are left due to the nature of this particular page. In particular, 3 of them are due to the embedded video, and I don't want to change anything about that.

### CODE REVIEW

File: about/what-is-fluent-bit.md [hunk 1]
```diff
@@ -4,14 +4,25 @@ description: Fluent Bit is a CNCF sub-project under the umbrella of Fluentd
...
+infrastructures. Managing telemetry data from various sources and formats can be a
+constant challenge, particularly when performance is a critical factor.
+
+Rather than serving as a drop-in replacement, Fluent Bit enhances the observability
+strategy for your infrastructure by adapting and optimizing your existing logging
+layer, and adding metrics and traces processing. Fluent Bit supports a
+vendor-neutral approach, seamlessly integrating with other ecosystems such as
+Prometheus and OpenTelemetry. Trusted by major cloud providers, banks, and companies
+in need of a ready-to-use telemetry agent solution, Fluent Bit effectively manages
+diverse data sources and formats while maintaining optimal performance.
```
Comment (lockewritesdocs) [inline about/what-is-fluent-bit.md:19]:
```suggestion
diverse data sources and formats while maintaining optimal performance and keeping
resource consumption low.
```

File: about/what-is-fluent-bit.md [hunk 2]
```diff
@@ -4,14 +4,25 @@ description: Fluent Bit is a CNCF sub-project under the umbrella of Fluentd
...
+
+Fluent Bit can be deployed as an edge agent for localized telemetry data handling or
+utilized as a central aggregator/collector for managing telemetry data across
+multiple sources and environments.
+
+[Fluent Bit](https://fluentbit.io) is designed with performance and low resource
+consumption in mind.
```
Comment (lockewritesdocs) [inline about/what-is-fluent-bit.md:26]:
```suggestion
```

**Review by lockewritesdocs — APPROVED** [review]
Suggested a few changes, but approving to keep the wheels of industry turning 🌐

File: about/what-is-fluent-bit.md (hunk 2 above)
Comment (esmerel) [reply about/what-is-fluent-bit.md:26]: yep, for sure!

**Pull request merged**